### ISSUE TITLE
Validate Softmax One and Quiet Attention

### ISSUE THREAD

**Issue by abrichr** [OpenAdaptAI/OpenAdapt#422]
### Feature request

https://www.evanmiller.org/attention-is-off-by-one.html

> I think a test could be hacked together fairly quickly: if you prefix every input context with a zero vector, and ensure that your neural network of choice doesn’t add any bias (including with the positional encoding), then the zero should pass through unaltered and will have the effect of adding unity to every subsequent softmax denominator; that way you won’t lose your mind mucking with gradient code. I think this could be done with a LLaMA model that uses fixed embeddings and a special prefix token, but I’d very much like to get back to my non-existent hobbies, and I’ve already spent more time on this problem than my therapist needs to know about.

> You’d still have to re-train the model, so don’t try this on an RPi just yet. But do let me know how those weight kurtoses and activation infinity norms are looking after a few runs. I’m thinking those numbers will make for a handsome table in a soon-to-be influential arXiV paper, either when those Qualcomm AI researchers step off the plane from Italy, or someone in an LLM hacker channel figures out biblatex, whichever happens first.

### Motivation

Computational efficiency, impact

(maybe not, see https://news.ycombinator.com/item?id=36854065)

**Comment by abrichr** [comment]
Closing as part of architecture transition to modular packages. Legacy codebase frozen at v0.46.0. See PR #960.
